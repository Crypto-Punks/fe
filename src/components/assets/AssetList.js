import React from 'react';
import PropTypes from 'prop-types';
import AssetItem from './AssetItem';

const AssetList = ({ activeCoins }) => {
  const coins = activeCoins.map(coin => {
    return <AssetItem key={coin.name} coin={coin} />; 
  });
  return (
    <ul>
      {coins}
    </ul>
  );
};

AssetList.propTypes = {
  activeCoins: PropTypes.arrayOf(PropTypes.shape({
    logo: PropTypes.string,
    name: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    value: PropTypes.string.isRequired
  })).isRequired
};

export default AssetList;
