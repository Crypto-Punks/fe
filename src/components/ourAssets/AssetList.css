.AssetList {
  position: relative;
  top: 10%;
  left: 25%;
  width: 50%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin: 0;
  
  a {
    text-decoration: none;
    font-family: abolition, sans-serif;
    font-weight: 400;
    font-style: normal;
  }

  a:visited {
    color: none;
  }

  li {
    position: relative;
    background-color: rgba(101, 134, 137, 0.845);
    margin: 0px;
    padding: 10px;
    height: 12vh;
    border-radius: .3em;
    list-style: none;
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: flex-start;
    transition: .2s ease-in-out;
    border: solid thin black;

    img {
      margin-top: 3%;
      width: 10%;
      height: 50%;
      border-radius: .5em;
    }

    p {
      color: white;
      font-family: abolition, sans-serif;
      font-size: 20pt;
      font-weight: 300;
      font-style: normal;
    }
    
    .Name {
      font-size: 20pt;
      margin-left: .5em;
      letter-spacing: .4em;
      font-weight: bold;
    }

    .Change {
      font-size: 15pt;
      margin-left: 1em;
      color: yellowgreen;
      letter-spacing: .4em;
    }

  }

  li:hover {
    transform: translateY(-0.2em);
    cursor: pointer;
    box-shadow: .5px .5px 30px 1px white;
  }
}


@media (max-width: 800px) {

  .AssetList {
    top: 40%;
    left: 0%;
    width: 80%;

    li {
      height: 10vh;
    }
  }
}

@media (max-width: 500px) {

  .AssetList {
    width: 80%;
    top: 60%;

    li {
      height: 3vh;
      border-radius: .75em;

      img{
        width: 10%;
      }

      p {
        font-size: 5pt;
        color: white;
      }
      
      .Name {
         margin-top: 0;
         font-size: 10pt;
      }

      .Change {
        font-size: 6pt;
        margin-left: 4em;
        letter-spacing: .3em;
        color: yellowgreen;
      }
    }
  }
}
