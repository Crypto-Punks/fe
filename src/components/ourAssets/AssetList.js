import React from 'react';
import PropTypes from 'prop-types';
import AssetItem from './AssetItem';
import styles from './AssetList.css';

const AssetList = ({ investedCoins }) => {
  const coins = investedCoins.map(coin => {
    return <AssetItem key={coin.name} activeCoin={coin} />; 
  });
  return (
    <ul className={styles.AssetList}>
      {coins}
    </ul>
  );
};

AssetList.propTypes = {
  investedCoins: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    logo: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    amount: PropTypes.number,
    price: PropTypes.string.isRequired
  })).isRequired
};

export default AssetList;
