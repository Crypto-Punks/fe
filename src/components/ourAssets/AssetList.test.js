import React from 'react';
import { shallow } from 'enzyme';
import AssetList from './AssetList';

const data = {
  investedCoins: []
};

describe('AssetList component', () => {
  it('renders AssetList', () => {
    const wrapper = shallow(<AssetList { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
