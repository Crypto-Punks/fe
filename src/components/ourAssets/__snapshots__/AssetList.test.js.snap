// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetList component renders AssetList 1`] = `
<ul
  className="AssetList"
/>
`;
